packages/core/nuxt-module/plugins/i18n-cookies.js

```diff
@@ -6,6 +6,7 @@ const i18nCookiesPlugin = ({ $cookies, i18n, app, redirect }) => {
   const navigator = isServer ? undefined : window.navigator;
   const acceptedLanguage = app.context.req?.headers?.['accept-language'] || navigator?.languages || '';
   const isRouteValid = !!app.context.route.name;
+  const autoRedirectByLocale = i18nOptions.autoRedirectByLocale ?? true;
```

Comment:
```suggestion
  const autoRedirectByLocale = i18nOptions.autoRedirectByLocale;
```

Reply:
I'm afraid that this change would implement BC as
```
!!undefined = false
```
and
```
!!(undefined ?? true) = true
```
and default should be `true`

Reply:
Yes, the default should be `true`.

`const autoRedirectByLocale = i18nOptions.autoRedirectByLocale;` would actually introduce a breaking change since not having this option defined in nuxt.config.js would result in disabling redirects (which should be on by default).

So not having this option defined in nuxt.config.js should still return true, that's why:
`undefined ?? true = true`

Having it set to false should return false:
`false ?? true = false`

And having it set to true should return true:
`true ?? true = true`

Reply:
OK, understood. Thanks!